tools/analyzer_plugin/lib/src/doc_utils/maturity.dart

```diff
@@ -0,0 +1,37 @@
+// Stolen directly from `package:analyzer/src/lint/linter.dart`
```

Comment:
Can we add proper attribution comments (including the license and everything) for code that was taken from linter?

We have an issue to add attributions, but since there's so much borrowed code and you know where it all came from, I think it' be a good idea to do it in this PR

Reply:
Added in https://github.com/Workiva/over_react/pull/525/commits/6515d0931a9140ed2c7a13a14ff4a42d00cce8c4